web/src/main/java/org/springframework/security/web/server/authentication/AuthenticationWebFilter.java

```diff
@@ -95,7 +112,12 @@ public AuthenticationWebFilter(ReactiveAuthenticationManager authenticationManag
 	private Mono<Void> authenticate(ServerWebExchange exchange,
 		WebFilterChain chain, Authentication token) {
 		WebFilterExchange webFilterExchange = new WebFilterExchange(exchange, chain);
-		return this.authenticationManager.authenticate(token)
+
+		ReactiveAuthenticationManager authenticationManager = this.authenticationManagerResolver
+				.resolve(exchange.getRequest())
+				.block();
```

Comment:
We actually can't introduce blocking operations into our reactive code. Instead, you'd want to map the result of this:

```java
return this.authenticationManagerResolver.resolve(exchange.getRequest())
		.map(authenticationManager -> authenticationManager.authenticate(token))
		.switchIfEmpty(...
```

Reply:
I had to cast `Mono<Authentication>` to `Authentication` to be able to use existing `onAuthenticationSuccess` as follow  

```
.flatMap(authentication -> onAuthenticationSuccess((Authentication) authentication, webFilterExchange))
``` 
Any thoughts?

Reply:
From the look of `test` failures, turns out I cannot do above 🤔

Reply:
At the moment, the only way that I can get the `flatMap` working is with `block` using existing `onAuthenticationSuccess` as follow 

```
.flatMap(authentication -> onAuthenticationSuccess(authentication.block(), webFilterExchange))
``` 
And since we should not use `blocking` operation in `reactive` then we might have to write a `onAuthenticationSuccess` that takes `Mono<Authentication>` or if there is a another way? 🤔